pkg/build/webhook/generic/generic.go

```diff
@@ -65,8 +66,13 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		}
 
 		var data api.GenericWebHookEvent
+		body, err = yaml.ToJSON(body)
```

Comment:
(And it means if someone specifies content-type json but supplies yaml, the payload will now be ignored because it'll catch an error)